ray-operator/controllers/ray/common/service.go

```diff
@@ -62,12 +62,16 @@ func BuildServiceForHeadPod(ctx context.Context, cluster rayv1.RayCluster, label
 	default_namespace := cluster.Namespace
 	default_type := cluster.Spec.HeadGroupSpec.ServiceType
 
-	defaultAppProtocol := utils.DefaultServiceAppProtocol
-	// `ports_int` is a map of port names to port numbers, while `ports` is a list of ServicePort objects
-	ports_int := getServicePorts(cluster)
+	// `serviceMapPortInfo` is a map of port names to port infos, while `ports` is a list of ServicePort objects
+	serviceMapPortInfo := getServicePorts(cluster)
 	ports := []corev1.ServicePort{}
-	for name, port := range ports_int {
-		svcPort := corev1.ServicePort{Name: name, Port: port, AppProtocol: &defaultAppProtocol}
+	for svc, portInfo := range serviceMapPortInfo {
+		appProtocol := utils.DefaultServiceAppProtocol
+		if portInfo.Protocol != "" {
+			appProtocol = portInfo.Protocol
+		}
+
+		svcPort := corev1.ServicePort{Name: svc, Port: portInfo.PortNumber, AppProtocol: &appProtocol}
```

Comment:
The `corev1.ServicePort` has two protocol fields. One is `Protocol` and the other one is `AppProtocol`.

To avoid confusion, I think we should rename the `portInfo.Protocol` to `portInfo.AppProtocol` here. That is:

```suggestion
		if portInfo.Protocol != "" {
			appProtocol = portInfo.AppProtocol
		}

		svcPort := corev1.ServicePort{Name: svc, Port: portInfo.PortNumber, AppProtocol: &appProtocol}
```

Reply:
Got it